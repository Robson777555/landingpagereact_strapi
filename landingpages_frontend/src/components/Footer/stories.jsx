import { Footer } from './Footer';

export default {
  title: 'Footer',
  component: Footer,
  args: {
    footerHtml:
      '<p><a href="https://github.com/Robson777555/landingpages_strapiv1">Feito com ‚ù§ por Robson Jobim</a></p>',
  },
};

export const Template = (args) => {
  return (
    <div>
      <Footer {...args} />
    </div>
  );
};
