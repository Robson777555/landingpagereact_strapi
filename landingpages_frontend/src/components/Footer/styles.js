import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default};
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    
    a{
      color: inherit;
      text-decoration: none;
      
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
  
`;
