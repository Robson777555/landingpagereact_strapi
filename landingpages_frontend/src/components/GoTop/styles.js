import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;

    /* Melhorias para mobile */
    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
    }

    /* Garante que seja clic√°vel em mobile */
    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;
