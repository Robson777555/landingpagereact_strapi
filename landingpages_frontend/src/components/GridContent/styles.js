import styled, { css } from 'styled-components';

export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    

  `}
`;

export const html = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    margin:${theme.spacings.xhuge} 0;
    text-align: center;
    
    pre {
      display: inline-block;
      
      margin: 0 auto;
    }

  `}
`;
