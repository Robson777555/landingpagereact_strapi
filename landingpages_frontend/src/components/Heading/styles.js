import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${css`
      @media (max-width: 768px) {
        font-size: ${theme.fonts.sizes.xlarge};
      }
    `}
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, $colorDark, $size, $uppercase }) => css`
    color: ${$colorDark ? theme.colors.primaryColor : theme.colors.white};
    background: transparent;
    padding: 1rem;
    
    ${titleSize[$size](theme)};
    ${titleCase($uppercase)};
  `}
`;
