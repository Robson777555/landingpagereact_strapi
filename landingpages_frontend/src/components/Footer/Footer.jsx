import P from 'prop-types';
import * as Styled from './styles';
import { SectionContainer } from '../SectionContainer/SectionContainer';

export const Footer = ({ footerHtml }) => {
  return (
    <Styled.Container>
      <SectionContainer>
        <div dangerouslySetInnerHTML={{ __html: footerHtml }} />
      </SectionContainer>
    </Styled.Container>
  );
};

Footer.propTypes = {
  footerHtml: P.string.isRequired,
};
