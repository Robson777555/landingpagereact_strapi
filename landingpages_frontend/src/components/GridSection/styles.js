import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';
export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${TextComponent} {
    margin-bottom: ${theme.spacings.xhuge};
   }
    
  `}
`;

export const Grid = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   counter-reset: grid-counter; 
   display: grid;
   grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
   gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${HeadingContainer} {
    position: relative;
    left: 5rem ;
   }

   ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
   }
    
  `}
`;
