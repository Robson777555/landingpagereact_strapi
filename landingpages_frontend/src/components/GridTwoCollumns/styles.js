import styled, { css } from 'styled-components';
import { Heading } from '../Heading/Heading';

export const Container = styled.div`
  ${({ theme, background, darkMode }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: ${theme.spacings.large};
    width: 100%;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    background: ${
      background
        ? darkMode
          ? theme.colors.darkBg
          : theme.colors.white
        : 'transparent'
    };
    color: ${darkMode ? theme.colors.white : theme.colors.primaryColor};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
      gap: ${theme.spacings.medium};
    }

    @media ${theme.media.lteSmall} {
      padding: ${theme.spacings.medium} ${theme.spacings.small};
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.small} ${theme.spacings.xsmall};
      gap: ${theme.spacings.small};
    }

    /* Regra específica para 325px de largura e até 852px de altura */
    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xsmall} ${theme.spacings.xxsmall || '0.4rem'};
      gap: ${theme.spacings.xsmall};
    }

    ${Heading} {
      color: ${darkMode ? theme.colors.white : theme.colors.black};
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.fonts.sizes.xxlarge};
      font-weight: 700;
      text-transform: uppercase;
      line-height: 1.2;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;

      @media ${theme.media.lteMedium} {
        font-size: ${theme.fonts.sizes.xlarge};
        margin-bottom: ${theme.spacings.small};
        white-space: normal;
      }

      @media ${theme.media.lteSmall} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.small};
      }

      @media ${theme.media.iphone} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.xsmall};
        white-space: nowrap;
        letter-spacing: -0.5px;
        padding: 0 ${theme.spacings.xsmall};
       
      }

      @media (max-width: 325px) and (max-height: 852px) {
        font-size: ${theme.fonts.sizes.medium};
        margin-bottom: ${theme.spacings.xsmall};
        letter-spacing: -0.3px;
        background: red;
      }
    }

    
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const TextComponent = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;

    @media ${theme.media.iphone} {
      padding: 0 ${theme.spacings.xsmall};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: 0 ${theme.spacings.xxsmall || '0.4rem'};
    }
  `}
`;

export const Image = styled.img`
  ${({ theme, darkMode }) => css`
    width: 100%;
    max-width: 400px;
    height: auto;
    border-radius: ${theme.spacings.xsmall};
    filter: ${darkMode ? 'brightness(0.85)' : 'none'};
    transition: filter 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      max-width: 350px;
    }

    @media ${theme.media.lteSmall} {
      max-width: 280px;
    }

    @media ${theme.media.iphone} {
      max-width: 220px;
      margin: 0 auto;
      display: block;
    }

    @media (max-width: 325px) and (max-height: 852px) {
      max-width: 180px;
    }
  `}
`;

export const ToggleButton = styled.button`
  ${({ theme }) => css`
    position: fixed;
    bottom: ${theme.spacings.medium};
    right: ${theme.spacings.medium};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    border-radius: 5px;
    cursor: pointer;
    z-index: 1000;
    transition: all 300ms ease-in-out;

    &:hover {
      background: ${theme.colors.secondaryColor};
      transform: scale(1.05);
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.xsmall} ${theme.spacings.small};
      bottom: ${theme.spacings.small};
      right: ${theme.spacings.small};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xxsmall || '0.4rem'} ${theme.spacings.xsmall};
      bottom: ${theme.spacings.xsmall};
      right: ${theme.spacings.xsmall};
      font-size: ${theme.fonts.sizes.xsmall};
      
    }
  `}
`;
