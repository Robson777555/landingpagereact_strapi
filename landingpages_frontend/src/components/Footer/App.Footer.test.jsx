import { screen } from '@testing-library/react';
import { Footer } from './Footer';
import { renderTheme } from '../../styles/render-theme';
import { describe, it, expect } from '@jest/globals';

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Footer footerHtml={'<h1>Olá</h1>'} />);
    expect(screen.getByRole('heading', { name: 'Olá' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
