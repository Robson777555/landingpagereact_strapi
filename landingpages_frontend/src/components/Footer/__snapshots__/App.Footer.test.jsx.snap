// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c2 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  text-align: center;
  font-size: 1.6rem;
  border-top: 0.1rem solid #DDDDDD;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

.c0 .c1 {
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div>
        <h1>
          Ol√°
        </h1>
      </div>
    </div>
  </footer>
</div>
`;
